python-package/lightgbm/basic.py

```diff
@@ -1000,12 +1013,13 @@ def construct(self):
                         ctypes.c_int(used_indices.shape[0]),
                         c_str(params_str),
                         ctypes.byref(self.handle)))
-                    self.data = self.reference.data
                     self.get_data()
                     if self.group is not None:
                         self.set_group(self.group)
                     if self.get_label() is None:
                         raise ValueError("Label should not be None.")
+                    if isinstance(self._predictor, _InnerPredictor) and self._predictor != self.reference._predictor:
```

Comment:
Can predictors be compared with `!=` operation?

Reply:
may be `is not` is better.